File: frontend/features/chat/types/api/deliverable.ts
import type { ToolExecutionResponse } from "./session";

export interface DeliverableResponse {
  id: string;
  session_id: string;
  kind: string;
  logical_name: string;
  latest_version_id?: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface DeliverableVersionResponse {
  id: string;
  session_id: string;
  run_id: string;
  deliverable_id: string;
  source_message_id?: number | null;
  version_no: number;
  file_path: string;
  file_name?: string | null;
  mime_type?: string | null;
  input_refs_json?: Record<string, unknown> | null;
  related_tool_execution_ids_json?: Record<string, unknown> | null;
  detection_metadata_json?: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}

export type DeliverableVersionToolExecutionResponse = ToolExecutionResponse;

File: backend/app/schemas/deliverable.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class DeliverableResponse(BaseModel):
    """Deliverable response schema."""

    id: UUID
    session_id: UUID
    kind: str
    logical_name: str
    latest_version_id: UUID | None = None
    status: str = "active"
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, populate_by_name=True)


class DeliverableVersionResponse(BaseModel):
    """Deliverable version response schema."""

    id: UUID
    session_id: UUID
    run_id: UUID
    deliverable_id: UUID
    source_message_id: int | None = None
    version_no: int
    file_path: str
    file_name: str | None = None
    mime_type: str | None = None
    input_refs_json: dict[str, Any] | None = None
    related_tool_execution_ids_json: dict[str, Any] | None = None
    detection_metadata_json: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
